standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/metrics/AcidMetricService.java

```diff
@@ -143,11 +220,33 @@ public static void updateMetricsFromShowCompact(ShowCompactResponse showCompactR
         Metrics.getOrCreateGauge(key).set(0);
       }
     }
+
+    Long numFailedComp = counts.get(TxnStore.FAILED_RESPONSE);
+    Long numNotInitiatedComp = counts.get(TxnStore.DID_NOT_INITIATE_RESPONSE);
+    Long numSucceededComp = counts.get(TxnStore.SUCCEEDED_RESPONSE);
+    if (numFailedComp != null && numNotInitiatedComp != null && numSucceededComp != null &&
+        ((numFailedComp + numNotInitiatedComp) / (numFailedComp + numNotInitiatedComp + numSucceededComp) >
+      MetastoreConf.getDoubleVar(conf, MetastoreConf.ConfVars.COMPACTOR_FAILED_COMPACTION_RATIO_THRESHOLD))) {
+      LOG.warn("Many compactions are failing. Check root cause of failed/not initiated compactions.");
+    }
+
     if (oldestEnqueueTime == Long.MAX_VALUE) {
       Metrics.getOrCreateGauge(COMPACTION_OLDEST_ENQUEUE_AGE).set(0);
     } else {
+      int oldestEnqueueAge = (int) ((System.currentTimeMillis() - oldestEnqueueTime) / 1000L);
       Metrics.getOrCreateGauge(COMPACTION_OLDEST_ENQUEUE_AGE)
-          .set((int) ((System.currentTimeMillis() - oldestEnqueueTime) / 1000L));
+          .set(oldestEnqueueAge);
+      if (oldestEnqueueAge >= MetastoreConf.getTimeVar(conf,
+          MetastoreConf.ConfVars.COMPACTOR_OLDEST_INITIATED_COMPACTION_TIME_THRESHOLD_WARNING, TimeUnit.SECONDS) &&
+          oldestEnqueueAge < MetastoreConf.getTimeVar(conf,
+              MetastoreConf.ConfVars.COMPACTOR_OLDEST_INITIATED_COMPACTION_TIME_THRESHOLD_ERROR, TimeUnit.SECONDS)) {
+        LOG.warn("Found compaction entry in compaction queue with an age of " + oldestEnqueueAge + " seconds. " +
+            "Check the time of last successful compaction and number of worker threads.");
```

Comment:
"Check the time of last successful compaction" -> I know this is part of the alert description but maybe this doesn't make sense...maybe just leave this at "Check the number of worker threads".. or "Consider increasing the number of worker threads"